### ISSUE TITLE
VTK v9 Approaching

### ISSUE THREAD

**Issue by banesullivan** [pyvista/pyvista#562]
[VTK v9 is on its way](https://discourse.vtk.org/t/vtk-9-release-coming/1859) 📦 🚀 (and see [milestone here](https://gitlab.kitware.com/vtk/vtk/-/milestones/15))

I figured I'd go ahead and bring this to everyone's attention as it looks like there will be changes that will affect PyVista. As it gets closer, let's use this issue to keep track of everything that needs to be adapted.

This will definitely further the need to set up a build pipeline on Azure to test PyVista against every public version of the VTK Python bindings hosted PyPI and conda-forge. I think we already need to do this in #517 as there are differences between VTK v8.1 and v8.2 in the numpy interface, but v9 will really make this necessary.

**Comment by banesullivan** [comment]
Status update: I'm having trouble the qt modules in VTKv9:

```py
>>> import vtk
>>> vtk.VTK_VERSION
'9.0.0'
>>> from vtk.qt.QVTKRenderWindowInteractor import QVTKRenderWindowInteractor
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/bane/anaconda3/envs/vtk9dev/lib/python3.7/site-packages/vtkmodules/qt/QVTKRenderWindowInteractor.py", line 66, in <module>
    from vtkmodules.vtkRenderingCore import vtkGenericRenderWindowInteractor, vtkRenderWindow
ImportError: cannot import name 'vtkGenericRenderWindowInteractor' from 'vtkmodules.vtkRenderingCore' (/Users/bane/anaconda3/envs/vtk9dev/lib/python3.7/site-packages/vtkmodules/vtkRenderingCore.so)
```

**Comment by akaszynski** [comment]
Is this from conda or did you build this?

**Comment by banesullivan** [comment]
conda-forge has it pre-built:

```
conda install -c conda-forge/label/vtk_dev vtk
```

**Comment by larsoner** [comment]
I have been building the wheels from scratch (takes about 6 lines and 10 minutes or so) and so far PyVista + Qt seems to [work okay aside from some rendering bugs with overlapping surfaces](https://github.com/mne-tools/mne-python/issues/7599)

**Comment by banesullivan** [comment]
Can you share those 6 lines?

**Comment by larsoner** [comment]
Prelease wheels now up at 

https://vtk.org/download/

**Comment by larsoner** [comment]
Hah! Didn't notice that gap. Feel free to comment here

https://gitlab.kitware.com/vtk/vtk/-/issues/17695

**Comment by larsoner** [comment]
Actually it looks like someone pointed it out already

**Comment by akaszynski** [comment]
I'm bumped for VTK wheels.  Not everyone uses anaconda and having the wheels is always helpful.

**Comment by banesullivan** [comment]
Even with the pre-built wheels, PyQt5 plotting is broken on Mac:

<img width="1920" alt="Screen Shot 2020-04-25 at 3 15 01 PM" src="https://user-images.githubusercontent.com/22067021/80288721-cd78fa80-8707-11ea-8d0e-22271abe8bd3.png">


The error stream I talked about in another issue is at least gone, but the renderer is transparent and not usable.

----

Per #628, I am inclined to switch entirely to using PySide. PyQt is outdated on conda-forge, has some sketchy licensing concerns, and is causing a headache with VTK9... I'm over it - PySide looks far more appealing to me now.

**Comment by larsoner** [comment]
Indeed it doesn't seem to work for me on macOS + VTK9rc3 + PyQt5 5.13.2 or 5.14.2.

**Comment by akaszynski** [comment]
@banesullivan I'm up for switching to `pyside` as well, or at least adding an option so we can support `pyside` and `pyqt`.  Since @GuillaumeFavelier is using `pyqt` I'd be hesitant to drop it entirely.  Plus, if you don't import pyqt, you don't incur the license restrictions.

**Comment by GuillaumeFavelier** [comment]
+1 for adding support for this. I think it's related to https://github.com/pyvista/pyvista/issues/628

**Comment by banesullivan** [comment]
FYI for those curious about the status of VTK9 on PyPI: they are waiting on https://github.com/pypa/pypi-support/issues/360

A bug in `setuptools` (see https://github.com/pypa/wheel/issues/203#issuecomment-62294421) causes the VTK wheel to triple in size

**Comment by larsoner** [comment]
FYI VTK 9 is now [up on PyPI](https://pypi.org/project/vtk/9.0.0/#files)

**Comment by panicbuttonvfx** [comment]
In case [this thread](https://discourse.vtk.org/t/vtk-9-pyqt-macos-no-rendering/3358) in VTK's support forum helps with the VTK 9 + Qt + macOS rendering issue. Workaround using pyvista involves to comment out a couple of lines in pyvista and adding them after `show()` in your app.
A side effect I've found is that when using area selection (e.g. `enable_cell_picking`), the plotter view turns black while dragging the selection and returns back to normal when stop dragging. Other than that, it all seems to work ok.

**Comment by MuellerSeb** [comment]
> FYI VTK 9 is now [up on PyPI](https://pypi.org/project/vtk/9.0.0/#files)

But linux-wheels are still missing although they increased the file-size limit (https://github.com/pypa/pypi-support/issues/360#issuecomment-630816647)

**Comment by banesullivan** [comment]
With the release of PyVista 0.25.1 and VTK9, I am closing this issue.

At this point, any problems warrant their own issue

**Comment by MuellerSeb** [comment]
Cool. I guess we have to wait for vtk 9.0.1 to get linux wheels.
